### PR TITLE
Add colors to flame chart.

### PR DESCRIPTION
**Pull request by kenzieschmoll**
Grabbed [Material color palettes](https://material.io/design/color/the-color-system.html#tools-for-picking-colors) Light Green 50 and Amber 50.

Colors can easily be changed - open to suggestions @InMatrix.

screen shot 2019-02-07 at 11 31 53 am

### CODE REVIEW

File: packages/devtools/lib/src/timeline/frame_flame_chart.dart [hunk 1]
```diff
@@ -10,6 +10,14 @@ import 'timeline_protocol.dart';
 // Switch this flag to true to dump the frame event trace to console.
 bool _debugEventTrace = false;
 
+// Amber 50 color palette from
+// https://material.io/design/color/the-color-system.html#tools-for-picking-colors.
+const cpuColorPalette = ['#FFECB3', '#FFE082', '#FFD54F', '#FFCA28'];
```
Comment (jacob314) [inline packages/devtools/lib/src/timeline/frame_flame_chart.dart]: Consider using the Color class forked from package:flutter for these. Having a class helps us port to flutter in the future and will help support light + dark themes.

File: packages/devtools/lib/src/timeline/frame_flame_chart.dart [hunk 2]
```diff
@@ -20,6 +28,20 @@ class FrameFlameChart extends CoreElement {
   CoreElement sectionTitles;
   CoreElement flameChart;
 
+  int cpuColorOffset = 0;
```
Comment (jacob314) [inline packages/devtools/lib/src/timeline/frame_flame_chart.dart]: nit: make cpuColorOffset private.

File: packages/devtools/lib/src/timeline/timeline_protocol.dart [hunk 3]
```diff
@@ -322,7 +322,14 @@ class TimelineFrame {
   final String id;
 
   /// Marks whether this frame has been added to the timeline.
-  final Completer<Null> addedToTimeline = Completer();
+  ///
```
Comment (jacob314) [inline packages/devtools/lib/src/timeline/timeline_protocol.dart]: nit: rebase so this change doesn't show up in this pull request.

File: packages/devtools/lib/src/timeline/frame_flame_chart.dart (hunk 1 above)
Comment (kenzieschmoll) [reply packages/devtools/lib/src/timeline/frame_flame_chart.dart]: Ah yes. Good idea - done.

File: packages/devtools/lib/src/timeline/frame_flame_chart.dart (hunk 2 above)
Comment (kenzieschmoll) [reply packages/devtools/lib/src/timeline/frame_flame_chart.dart]: done.

File: packages/devtools/lib/src/timeline/timeline_protocol.dart (hunk 3 above)
Comment (kenzieschmoll) [reply packages/devtools/lib/src/timeline/timeline_protocol.dart]: odd, i tried to rebase but I must have messed up somewhere. Good to go now.

File: packages/devtools/lib/src/timeline/frame_flame_chart.dart [hunk 4]
```diff
@@ -20,6 +40,20 @@ class FrameFlameChart extends CoreElement {
   CoreElement sectionTitles;
   CoreElement flameChart;
 
+  int _cpuColorOffset = 0;
+  Color get cpuColor {
+    final color = cpuColorPalette[_cpuColorOffset % cpuColorPalette.length];
+    _cpuColorOffset++;
```
Comment (devoncarew) [inline packages/devtools/lib/src/timeline/frame_flame_chart.dart]: It looks like this getter is modifying state for the object. You may want to use a regular method, and change the name to something that indicates that multiple calls will change state; `Color nextCpuColor() {...`?

**Comment by devoncarew** [comment]
The bot failure is:

```
01:48 +24 -1: test/inspector_controller_test.dart: inspector controller tests hotReload [E]
  TimeoutException after 0:00:30.000000: Test timed out after 30 seconds.
  package:test_api  Invoker._onRun.<fn>.<fn>.<fn>
```

I don't know if this is a test that is not progressing, or is just running close to the default time limit. It may be worth doubling the test timeout time for this test (or the group) (`, timeout: new Timeout.factor(2))`) to see if that unflakes the test.

File: packages/devtools/lib/src/timeline/frame_flame_chart.dart (hunk 4 above)
Comment (kenzieschmoll) [reply packages/devtools/lib/src/timeline/frame_flame_chart.dart]: done.

**Comment by kenzieschmoll** [comment]
Increasing test timeout seems to have fixed it 👍

**Pull request merged**